### PR TITLE
cleanup some backend warnings. (can wait until next sprint if necessary).

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: source/backend/api/Areas/Admin/Controllers/ClaimController.cs [hunk 1]
```diff
@@ -95,66 +93,6 @@ public IActionResult GetClaim(Guid key)
             var claim = _mapper.Map<Model.ClaimModel>(entity);
             return new JsonResult(claim);
         }
-
```
Comment (devinleighsmith) [inline source/backend/api/Areas/Admin/Controllers/ClaimController.cs:98]: dead controller methods, we do not allow roles/claims to be edited in MOTI PIMS.

File: source/backend/api/Areas/Reports/Controllers/LeaseController.cs [hunk 2]
```diff
@@ -74,7 +74,7 @@ public LeaseController(ILookupRepository lookupRepository, ILeaseRepository leas
         /// <returns></returns>
         [HttpGet]
         [HasPermission(Permissions.LeaseView)]
-        [Produces(ContentTypes.CONTENT_TYPE_CSV, ContentTypes.CONTENT_TYPE_EXCELX)]
+        [Produces(ContentTypes.CONTENTTYPECSV, ContentTypes.CONTENTTYPEEXCELX)]
```
Comment (devinleighsmith) [inline source/backend/api/Areas/Reports/Controllers/LeaseController.cs:77]: Looking back on this I'm not sure this was such a good idea.

File: source/backend/api/Configuration/ProgramOptions.cs [hunk 3]
```diff
@@ -42,7 +43,7 @@ public string[] ToArgs()
             {
                 this.Urls,
                 this.Environment,
-                this.HttpsPort?.ToString(),
+                this.HttpsPort?.ToString(CultureInfo.InvariantCulture),
```
Comment (devinleighsmith) [inline source/backend/api/Configuration/ProgramOptions.cs:46]: using invariant culture on calculations/non-display values

File: source/backend/api/Controllers/ModelController.cs [hunk 4]
```diff
@@ -1,56 +0,0 @@
...
-namespace Pims.Api.Areas.Dummy.Controllers
-{
-    /// <summary>
-    /// ModelController class, provides endpoints for interacting with leases.
-    /// </summary>
-    [ApiController]
-    [ApiVersion("2.0")]
-    [Area("models")]
-    [Route("[area]")]
-    public class ModelController : ControllerBase
```
Comment (devinleighsmith) [inline source/backend/api/Controllers/ModelController.cs:13]: this looks like a test controller that was forgotten about.

File: source/backend/api/Helpers/Exceptions/ApiHttpRequestException.cs [hunk 5]
```diff
@@ -60,7 +60,7 @@ public ApiHttpRequestException(string message, Exception innerException, HttpSta
         public ApiHttpRequestException(HttpResponseMessage response)
             : base($"HTTP Request '{response.RequestMessage.RequestUri}' failed", null, response?.StatusCode)
         {
-            this.Response = response ?? throw new ArgumentNullException(nameof(response)); // NOSONAR
```
Comment (devinleighsmith) [inline source/backend/api/Helpers/Exceptions/ApiHttpRequestException.cs:63]: evidently there is no way the argumentNullException can be triggered here.

File: source/backend/api/Helpers/Middleware/ErrorHandlingMiddleware.cs [hunk 6]
```diff
@@ -84,11 +84,6 @@ private async Task HandleExceptionAsync(HttpContext context, Exception ex)
                 code = HttpStatusCode.Unauthorized;
                 message = "The authentication token is invalid.";
             }
-            else if (ex is SecurityTokenValidationException)
```
Comment (devinleighsmith) [inline source/backend/api/Helpers/Middleware/ErrorHandlingMiddleware.cs:87]: duplicate, likely merge artifact.

File: source/backend/api/Helpers/Middleware/LogRequestMiddleware.cs [hunk 7]
```diff
@@ -67,8 +67,8 @@ public async Task Invoke(HttpContext context)
                     _logger.LogInformation($"HTTP Request {context.Request.Method} user:{context.User.GetDisplayName()} {context.Request.Scheme}://{context.Request.Host}{context.Request.Path}{context.Request.QueryString}");
                 }
 
-                _logger.LogDebug($"HTTP Request {context.Request.Method} user:{context.User.GetDisplayName()} {context.Request.Scheme}://{context.Request.Host}{context.Request.Path}{context.Request.QueryString}");
-                _logger.LogTrace(string.IsNullOrEmpty(body) ? string.Empty : $"{System.Environment.NewLine}Body: {body}");
+                _logger.LogDebug("HTTP Request {context.Request.Method} user:{context.User.GetDisplayName()} {context.Request.Scheme}://{context.Request.Host}{context.Request.Path}{context.Request.QueryString}", context.Request.Method, context.User.GetDisplayName(), context.Request.Scheme, context.Request.Host, context.Request.Path, context.Request.QueryString);
```
Comment (devinleighsmith) [inline source/backend/api/Helpers/Middleware/LogRequestMiddleware.cs:70]: evidently this is the preferred format for log messages, to make it easier to search for related log messages.

File: source/backend/api/Helpers/Swagger/ConfigureSwaggerForOpenApi.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+using System.Globalization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using Swashbuckle.AspNetCore.SwaggerUI;
+
+namespace Pims.Api.Helpers.Swagger
```
Comment (devinleighsmith) [inline source/backend/api/Helpers/Swagger/ConfigureSwaggerForOpenApi.cs:11]: This class allows the config from appsettings to control the display of some fields in swagger.

File: source/backend/core/Http/IHttpRequestClient.cs [hunk 9]
```diff
@@ -20,12 +20,6 @@ Task<HttpResponseMessage> SendJsonAsync<T>(string url, HttpMethod method, HttpRe
         Task<HttpResponseMessage> PostJsonAsync<T>(string url, T data = null)
             where T : class;
 
-        Task<HttpResponseMessage> PutJsonAsync<T>(string url, T data = null)
```
Comment (devinleighsmith) [inline source/backend/core/Http/IHttpRequestClient.cs:23]: removing unused methods.

File: source/backend/dal/Repositories/Interfaces/IRoleRepository.cs [hunk 10]
```diff
@@ -12,21 +12,5 @@ public interface IRoleRepository : IRepository<PimsRole>
         Paged<PimsRole> GetPage(int page, int quantity, string name = null);
 
         PimsRole GetByKey(Guid key);
-
-        PimsRole GetByName(string name);
```
Comment (devinleighsmith) [inline source/backend/dal/Repositories/Interfaces/IRoleRepository.cs:16]: unused methods.

File: source/backend/dal/Repositories/RoleRepository.cs [hunk 11]
```diff
@@ -75,129 +74,6 @@ public PimsRole GetByKey(Guid key)
...
                 .FirstOrDefault(u => u.RoleUid == key) ?? throw new KeyNotFoundException();
         }
-
-        /// <summary>
-        /// Get the role with the specified name.
-        /// </summary>
-        /// <param name="name"></param>
-        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
-        /// <returns></returns>
-        public PimsRole GetByName(string name)
```
Comment (devinleighsmith) [inline source/backend/dal/Repositories/RoleRepository.cs:85]: unused, from citz pims

**Review by asanchezr — APPROVED** [review]

**Pull request merged**